apps/voting/app/src/math-utils.js

```diff
@@ -38,10 +39,88 @@ export function formatNumber(num, decimals = 2, { truncate = true } = {}) {
     : formattedNumber
 }
 
-export function percentageList(values, digits = 0) {
+export function percentageList(values) {
   return scaleBNValuesSet(values).map(value => value.toNumber())
 }
 
+// Format a percentage from BN values.
+// `value` divided by `pctBase` must be in the [0, 1] range.
+export function formatBnPercentage(
+  value,
+  pctBase,
+  { digits = 2, suffix = '%' } = {}
+) {
+  let basePrecision = 10 ** digits
+
+  // Tolerate having too many digits by correcting the value.
+  if (basePrecision > Number.MAX_SAFE_INTEGER) {
+    digits = Math.floor(Math.log(Number.MAX_SAFE_INTEGER) / Math.log(10))
```

Comment:
Not sure about this:

1. I think we should make this max precision value be a constant
2. This value is actually only `15`, and and I think we should at least support 18 decimals (since that's what tokens use, as well as our contract's `pctBase` exponent).

It's unlikely someone will want something this large, but I'm also wondering if we shouldn't just cap the precision at the `pctBase`'s exponent (since it'll just be 0s after that anyway).

Reply:
> I think we should make this max precision value be a constant

Why? This is already on the Javascript `Number` object, and so having this as a constant seems a bit redundant to me.

> This value is actually only 15, and and I think we should at least support 18 decimals (since that's what tokens use, as well as our contract's pctBase exponent).

This is true, and relates to #1177 in terms of having an utility to safely scale up normal numbers to Bigints.

Reply:
> Why? This is already on the Javascript Number object, and so having this as a constant seems a bit redundant to me.

Because `Math.floor(Math.log(Number.MAX_SAFE_INTEGER) / Math.log(10))` is pure and doesn't need to be calculated every time